domain/src/test/java/org/oppia/android/domain/feedbackreporting/FeedbackReportManagementControllerTest.kt

```diff
@@ -0,0 +1,445 @@
+    reportsStore.observeForever(mockReportsStoreObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockReportsStoreObserver, atLeastOnce()).onChanged(reportStoreResultCaptor.capture())
+    val reportsList = reportStoreResultCaptor.value.getOrThrow().reportsList
+    val report = reportsList.get(0)
+    assertThat(report).isEqualTo(earlierCrashReport)
+    feedbackReportManagementController.removeFirstCachedReport()
+
+    val emptyReportsStore = feedbackReportManagementController.getFeedbackReportStore().toLiveData()
+    emptyReportsStore.observeForever(mockReportsStoreObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(mockReportsStoreObserver, atLeastOnce()).onChanged(reportStoreResultCaptor.capture())
+
+    val reportsCount = reportStoreResultCaptor.value.getOrThrow().reportsCount
+    assertThat(reportsCount).isEqualTo(0)
+  }
+
+  @Test
+  fun testController_removeCachedReports_removedInOrder() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    feedbackReportManagementController.submitFeedbackReport(earlierCrashReport)
+    feedbackReportManagementController.submitFeedbackReport(laterSuggestionReport)
+    val reportsStore = feedbackReportManagementController.getFeedbackReportStore().toLiveData()
+    reportsStore.observeForever(mockReportsStoreObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockReportsStoreObserver, atLeastOnce()).onChanged(reportStoreResultCaptor.capture())
+    val removedReport = reportStoreResultCaptor.value.getOrThrow().getReports(0)
+    feedbackReportManagementController.removeFirstCachedReport()
+
+    val updatedReportsStore = feedbackReportManagementController.getFeedbackReportStore()
+      .toLiveData()
+    updatedReportsStore.observeForever(mockReportsStoreObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(mockReportsStoreObserver, atLeastOnce()).onChanged(reportStoreResultCaptor.capture())
+
+    val nextReport = reportStoreResultCaptor.value.getOrThrow().getReports(0)
+    assertThat(nextReport).isNotEqualTo(removedReport)
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  private fun setUpApplicationForContext() {
+    // The package manager is shadowed in order to specify the package version name and package
+    // version code so that the controller can correctly fetch and set these values in the feedback
+    // report.
+    val packageManager = Shadows.shadowOf(context.packageManager)
+    val applicationInfo =
+      ApplicationInfoBuilder.newBuilder()
+        .setPackageName(context.packageName)
+        .build()
+    val packageInfo =
+      PackageInfoBuilder.newBuilder()
+        .setPackageName(context.packageName)
+        .setApplicationInfo(applicationInfo)
+        .build()
+    packageInfo.versionName = "1.0"
+    packageInfo.versionCode = 1
```

Comment:
Should we try `packageInfo.longVersionCode = 1` as `versionCode` is deprecated. 
Check `PackageInfoCompat` class.

Reply:
Done, updated this to be a `Long`. Fixed the proto and extension class as well.